File: ccif/src/java/other/org/himalay/ccif/ProgramInfo.java
// Copyright (2013) Krishna C Tripathi. All rights reserved.
// 
// You are not allowed to read/copy/distribute following code without explicit written authorization from Krishna C Tripathi
//
package org.himalay.ccif ;

import java.io.DataInputStream;
import java.io.DataOutputStream;
import java.io.IOException;
import java.lang.Exception;
import java.io.ByteArrayInputStream;
import java.io.PrintStream;
import org.himalay.msgs.runtime.Created;

import org.himalay.msgs.runtime.*;
@Created(date = "Fri Jan 15 01:32:55 EST 2016")

public  class ProgramInfo extends BinStruct implements PublicBinMsg{




    // members variables
    // bf1
    public BitField_16 bf1 ;
    // programInfo
    public CaInfo programInfo ;

        public ProgramInfo () // throws Exception
    {
            init();
    }
    
    private void init()
    {
        // Initialize bf1
        bf1 = new BitField_16();
        // Initialize programInfo
        programInfo = new CaInfo();
    }

    public int readNoHeader(DataInputStream istream) throws IOException 
    {
    
     return read(istream);
       }
   
    public int read(DataInputStream istream) throws IOException 
    {
    preRead();
        int retVal= 0;

                     // read bf1
        retVal += bf1.read(istream);
        // read programInfo
            {        int iCount    = getProgramInfoLength()+(0);        byte[] ba        = new byte[iCount];istream.readFully(ba);ByteArrayInputStream baTemp= new ByteArrayInputStream(ba);        retVal+= programInfo.read(new DataInputStream(baTemp));}

        postRead();
        return retVal;
    }


    public int write(DataOutputStream ostream) throws IOException 
    {
    preWrite();
        int retVal= 0;

        
        { /** fix dependent sizes for programInfo **/ if (programInfo != null )setProgramInfoLength((short)programInfo.getSize()); }
    
                // write bf1
        ostream.writeShort(bf1.getValue()); retVal +=2;
        // write programInfo
        if (programInfo!=null)retVal +=programInfo.write(ostream);
postWrite();
        return retVal;
    }
    
    public int dump(DumpContext dc) throws IOException 
    {
        dc.indent();dc.getPs().print("ProgramInfo\n");
    dc.increaseIndent();
        int retVal= 0;
        // write bf1
        {dc.indent();dc.getPs().print("reserved2: ");dc.getPs().println(BitField_8.toDisplayString(getReserved2(),4));dc.indent();dc.getPs().print("programInfoLength: ");dc.getPs().println(BitField_8.toDisplayString(getProgramInfoLength(),12));}
        // write programInfo
        if ( programInfo != null ) {dc.indent();dc.getPs().println("programInfo") ;retVal +=programInfo.dump(dc);}
dc.decreaseIndent();
        return retVal;
    }


        // Getter for bf1
    //public BitField_16 getBf1()
    //{
    //    return bf1 ;
    //}

    
    // Setter for bf1
    //public void setBf1(BitField_16 val)
    //{
    //    this.bf1= val;
    //}
    // Getter for programInfo
    //public CaInfo getProgramInfo()
    //{
    //    return programInfo ;
    //}

    
    // Setter for programInfo
    //public void setProgramInfo(CaInfo val)
    //{
    //    this.programInfo= val;
    //}


    public int getReserved2( )
    {
        return ( bf1.getValue() & 0x0000f000 ) >> 12 ;
    }
    
    
    public void setReserved2(int val)
    {
        bf1.setValue ( (bf1.getValue() & 0xffff0fff) | ( (val << 12 ) & 0x0000f000));
    }
    
    
    public int getProgramInfoLength( )
    {
        return ( bf1.getValue() & 0x00000fff ) >> 0 ;
    }
    
    
    public void setProgramInfoLength(int val)
    {
        bf1.setValue ( (bf1.getValue() & 0xfffff000) | ( (val << 0 ) & 0x00000fff));
    }
    
    
    public int getSize() throws IOException
    {
       DataOutputStream dos= new DataOutputStream(new NullStream());
       return this.write(dos);
    }
    
        

}

// End of code

File: libccif/src/bp/ProgramInfo.cpp
//**//
//@Created(date = "Mon Feb 01 22:30:11 EST 2016")
// Copyright (2012) Krishna C Tripathi. All rights reserved.
//
// You are not allowed to read/copy/distribute following code without explicit written authorization from Krishna C Tripathi
//
//**//
#include "stdio.h"
#include "ProgramInfo.h"

#ifdef org_himalay_ccif_ProgramInfo__USE_SMART_PTR
#define ArrayList(x) SmartPtrList<x>
#define ArrayList_iterator(x) SmartPtrList<x>::iterator
#include "SmartPtrList.h"
#else
#define ArrayList(x) BinMessagePtrList
#define ArrayList_iterator(x) BinMessagePtrList::iterator
#include "BinMessagePtrList.h"
#endif

// Namespace
namespace org {
namespace himalay {
namespace ccif {

using namespace ::org::himalay::msgs::runtime;

// Constructor
ProgramInfo::ProgramInfo() {
    references= 0;


    // bf1
    bf1 = 0;
    // programInfo
    org_himalay_ccif_CaInfo__NEW( programInfo);
}
// Destructor
ProgramInfo::~ProgramInfo() {
    // bf1

    // programInfo
    org_himalay_ccif_CaInfo__DELETE( programInfo);
}

int ProgramInfo::readNoHeader(DataInputStream&  istream ) { // throws IOException
    int retVal = 0;

    // bf1
    bf1= istream.readUI16();
    retVal += 2;
    // programInfo
    {
        ui8* ba= new ui8[getProgramInfoLength()+(0)];
        int iCount = istream.read(ba,getProgramInfoLength()+(0));
        ByteArrayDataInputStream* bais= new ByteArrayDataInputStream(ba,0,iCount );
        retVal += programInfo->read(*bais);
        delete bais;
    }
    return retVal;
}

int ProgramInfo::read(DataInputStream&  istream ) { // throws IOException
    int retVal = 0;
    // read bf1
    bf1= istream.readUI16();
    retVal += 2;
    // read programInfo
    {
        ui8* ba= new ui8[getProgramInfoLength()+(0)];
        int iCount = istream.read(ba,getProgramInfoLength()+(0));
        ByteArrayDataInputStream* bais= new ByteArrayDataInputStream(ba,0,iCount );
        retVal += programInfo->read(*bais);
        delete bais;
    }

    return retVal;
}

int ProgramInfo::write(DataOutputStream& ostream) { // throws IOException

{   /** fix dependent sizes for programInfo **/ if ( programInfo != NULL ) {
            setProgramInfoLength((short)programInfo->getSize());
        }
    }

    int retVal= 0;
    // write bf1
    ostream.writeUI16(bf1);
    retVal +=2;
    // write programInfo
    if ( programInfo != NULL ) {
        retVal +=programInfo->write(ostream);
    }

    return retVal;
}

int ProgramInfo::dump(DumpContext& dc) { // throws IOException
    dc.indent();
    dc.getPs().print("ProgramInfo\n");
    dc.increaseIndent();
    int retVal= 0;
    // write bf1
    {
        dc.indent();
        dc.getPs().print("reserved2: ");
        dc.getPs().printlnBin(getReserved2(),4);
        dc.indent();
        dc.getPs().print("programInfoLength: ");
        dc.getPs().printlnBin(getProgramInfoLength(),12);
    }
    // write programInfo
    if ( programInfo != NULL ) {
        dc.indent();
        dc.getPs().println("programInfo") ;
        retVal +=programInfo->dump(dc);
    }
    dc.decreaseIndent();
    return retVal;
}


// Getter for bf1
//ui16 ProgramInfo::getBf1()
//{
//return this->bf1;
//}

// Setter for bf1
//void ProgramInfo::setBf1(ui16 val)
//{
//this->bf1= val;
//}

// Getter for programInfo
//org_himalay_ccif_CaInfo__PTR_TYPE ProgramInfo::getProgramInfo()
//{
//return this->programInfo;
//}

// Setter for programInfo
//void ProgramInfo::setProgramInfo(org_himalay_ccif_CaInfo__PTR_TYPE val)
//{
//this->programInfo= val;
//}

int ProgramInfo::getReserved2( ) {
    return ( this->bf1 & 0x0000f000 ) >> 12 ;
}

void ProgramInfo::setReserved2(int val) {
    bf1=(  (this->bf1 & 0xffff0fff) | ( (val << 12 ) & 0x0000f000));
}

int ProgramInfo::getProgramInfoLength( ) {
    return ( this->bf1 & 0x00000fff ) >> 0 ;
}

void ProgramInfo::setProgramInfoLength(int val) {
    bf1=(  (this->bf1 & 0xfffff000) | ( (val << 0 ) & 0x00000fff));
}

int ProgramInfo::getSize() { // throws IOException
    int retVal = 0;

    return retVal;
}

#ifdef org_himalay_ccif_ProgramInfo__USE_SMART_PTR
void intrusive_ptr_add_ref(org::himalay::ccif::ProgramInfo* p) {
    p->increaseRef();
};
void intrusive_ptr_release(org::himalay::ccif::ProgramInfo* p) {
    p->decreaseRef();
};;
#endif


}
}
}

// End of code